### PR TITLE
Provide draco-7_8_0.

### PR DESCRIPTION
**Pull request by KineticTheory**
+ Also provide a patchfile for draco-7_6_0 to support CrayPE builds.
+ Version 7.8.0 has a new variant `+caliper`.
+ Sort dependencies alphabetically after grouping by required and optional.

### CODE REVIEW

File: var/spack/repos/builtin/packages/draco/package.py [hunk 1]
```diff
@@ -64,13 +68,14 @@ class Draco(CMakePackage):
...
     conflicts('+cuda', when='@:7.6.99')
+    conflicts('+caliper', when='@:7.7.99')
 
     # Fix python discovery.
     patch('d710.patch', when='@7.1.0^python@3:')
     patch('d710-python2.patch', when='@7.1.0^python@2.7:2.99')
     patch('d730.patch', when='@7.3.0:7.3.99')
     patch('d740.patch', when='@7.4.0:7.4.99')
-    patch('smpi.patch', when='@:7.6.99')
+    patch('d760-cray.patch', when='@7.6.0')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/draco/package.py:78]: Is `smpi.patch` used at all now? should we delete it from the repository?

**Review by alalazo — COMMENTED** [review]
Just a question on the new patch, otherwise LGTM.

File: var/spack/repos/builtin/packages/draco/package.py (hunk 1 above)
Comment (KineticTheory) [reply var/spack/repos/builtin/packages/draco/package.py:78]: > Is `smpi.patch` used at all now? should we delete it from the repository?

@alalazo That looks like a bug on my part.  Let me recheck the whole file to make sure I didn't mess something else up.  Thanks for noticing!

Comment (KineticTheory) [reply var/spack/repos/builtin/packages/draco/package.py:78]: @alalzo Indeed, older versions of draco are no longer supported for `^spectrum-mpi` and newer versions work without the patch.  I've deleted the excess file in my updated PR.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**